package/src/components/MessageInput/InputButtons.tsx

```diff
@@ -1,56 +1,66 @@
-import React, { useCallback } from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { StyleSheet, View } from 'react-native';
 
+import { CustomDataManagerState, TextComposerState } from 'stream-chat';
+
+import { useMessageComposer } from '../../contexts/messageInputContext/hooks/useMessageComposer';
 import {
   MessageInputContextValue,
   useMessageInputContext,
 } from '../../contexts/messageInputContext/MessageInputContext';
 import { useOwnCapabilitiesContext } from '../../contexts/ownCapabilitiesContext/OwnCapabilitiesContext';
 import { useTheme } from '../../contexts/themeContext/ThemeContext';
-
-const styles = StyleSheet.create({
-  attachButtonContainer: { paddingRight: 5 },
-});
+import { useStateStore } from '../../hooks/useStateStore';
 
 export type InputButtonsProps = Partial<InputButtonsWithContextProps>;
 
 export type InputButtonsWithContextProps = Pick<
   MessageInputContextValue,
   | 'AttachButton'
   | 'CommandsButton'
-  | 'giphyActive'
   | 'hasCameraPicker'
   | 'hasCommands'
   | 'hasFilePicker'
   | 'hasImagePicker'
-  | 'hasText'
   | 'MoreOptionsButton'
-  | 'openCommandsPicker'
   | 'selectedPicker'
-  | 'setShowMoreOptions'
-  | 'showMoreOptions'
   | 'toggleAttachmentPicker'
 >;
 
+const textComposerStateSelector = (state: TextComposerState) => ({
+  text: state.text,
+});
+
+const customComposerDataSelector = (state: CustomDataManagerState) => ({
+  command: state.custom.command,
+});
+
 export const InputButtonsWithContext = (props: InputButtonsWithContextProps) => {
   const {
     AttachButton,
     CommandsButton,
-    giphyActive,
     hasCameraPicker,
     hasCommands,
     hasFilePicker,
     hasImagePicker,
-    hasText,
     MoreOptionsButton,
-    openCommandsPicker,
-    setShowMoreOptions,
-    showMoreOptions,
   } = props;
+  const { customDataManager, textComposer } = useMessageComposer();
+  const { text } = useStateStore(textComposer.state, textComposerStateSelector);
+  const { command } = useStateStore(customDataManager.state, customComposerDataSelector);
+  const [showMoreOptions, setShowMoreOptions] = useState(true);
```

Comment:
```suggestion
const showMoreOptions = useMemo(() => Boolean(text.length); 
```
should be more than enough here, that way you neither need to pass a default value nor do you need to rely on `React` state.

Reply:
Well, I would still need the `setShowMoreOptions` to expand the input buttons when text is there on click of the more options button. But I will check how I can optimize this.

Reply:
Ah, I see - so it's used elsewhere too. Then, you can ignore this comment. Resolving it.

Reply:
Made it a bit better by introducing `hasText` so that the effect doesn't have to run multiple times. :)